lib/collective.lib.js

```diff
@@ -49,6 +49,11 @@ export const getCollectiveMainTag = (hostCollectiveId = null, tags = [], type, s
   return CollectiveCategory.COLLECTIVE;
 };
 
+export const expenseSubmissionAllowed = (collective, user) => {
```

Comment:
Though frontend permissions check are usually deprecated in favor of API permissions checks (eg. with the `permissions` fields on expenses and transactions), the ones that we have in the frontend are mostly centralized in https://github.com/opencollective/opencollective-frontend/blob/8d9fc3127212106822b243c17bb6f2deff2855ec/lib/LoggedInUser.js. I think it would be great to follow the same pattern and have something like `LoggedInUser.canSubmitExpense(collective)`.

Reply:
@Betree : Notice that this function should handle the case where `LoggedInUser` is null (because it can be called when the user is not logged in). In that sense, my thinking is that it makes sense to pass the `user` object rather than trying to call the function upon the `LoggedInUser`. Does that make sense? Or do you have an alternative suggestion. 😄

Reply:
That makes sense, if `collective?.settings.disablePublicExpenseSubmission` is not set submission should be allowed even if the user does not exist.